extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js

```diff
@@ -0,0 +1,1496 @@
+      this.state.contentTree.duration = asset.content.duration;
+      this.state.contentTree.promo_image = asset.content.posterImages[0].url;
+
+      this.renderSkin({"contentTree": this.state.contentTree});
+    },
+
+    onVolumeChanged: function (event, newVolume) {
+      if (newVolume <= 0) {
+        this.state.volumeState.muted = true;
+        this.state.volumeState.volume = 0;
+      } else {
+        this.state.volumeState.muted = false;
+        this.state.volumeState.volume = newVolume;
+      }
+      this.renderSkin();
+    },
+
+    resetUpNextInfo: function (purge) {
+      if (purge) {
+        this.state.upNextInfo.upNextData = null;
+      }
+      this.state.upNextInfo.countDownFinished = false;
+      this.state.upNextInfo.countDownCancelled = false;
+    },
+
+    onPlayheadTimeChanged: function(event, currentPlayhead, duration, buffered, startEnd, videoId) {
+      if (videoId == OO.VIDEO.MAIN) {
+        this.state.mainVideoPlayhead = currentPlayhead;
+        this.state.mainVideoDuration = duration;
+        this.state.mainVideoBuffered = buffered;
+      }
+      else if (videoId == OO.VIDEO.ADS) {
+        //adVideoDuration is only used in adPanel ad marquee
+        this.state.adVideoDuration = duration;
+      }
+      this.state.duration = duration;
+
+      // lower skin z-index if Chrome auto-pauses flash content
+      if(!this.state.autoPauseDisabled && Utils.isChrome() && this.state.mainVideoMediaType != CONSTANTS.MEDIA_TYPE.HTML5) {
+        var skinElement = $("#"+this.state.elementId+" .oo-player-skin");
+        if(currentPlayhead == 0 && this.state.playerState == CONSTANTS.STATE.PLAYING) {
+          skinElement.addClass('oo-z-index-auto');
+        } else {
+          skinElement.removeClass('oo-z-index-auto');
+          this.state.autoPauseDisabled = true;
+        }
+      }
+
+      // The code inside if statement is only for up next, however, up next does not apply to Ad screen.
+      // So we only need to update the playhead for ad screen.
+      if (this.state.screenToShow !== CONSTANTS.SCREEN.AD_SCREEN ) {
+        if (this.skin.props.skinConfig.upNext.showUpNext) {
+          if (!(Utils.isIPhone() || (Utils.isIos() && this.state.fullscreen))){//no UpNext for iPhone or fullscreen iPad
+            this.showUpNextScreenWhenReady(currentPlayhead, duration);
+          }
+        } else if (this.state.playerState === CONSTANTS.STATE.PLAYING) {
+          this.state.screenToShow = CONSTANTS.SCREEN.PLAYING_SCREEN;
+        } else if (this.state.playerState === CONSTANTS.STATE.PAUSE) {
+          this.state.screenToShow = CONSTANTS.SCREEN.PAUSE_SCREEN;
+        }
+      }
+      if (!this.state.seeking) {
+        this.skin.updatePlayhead(currentPlayhead, duration, buffered);
+      } else {
+        this.state.queuedPlayheadUpdate = [currentPlayhead, duration, buffered];
+      }
+    },
+
+    showUpNextScreenWhenReady: function(currentPlayhead, duration) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 11 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)